### ISSUE TITLE
A change in a top-level statement or an addition of a new top-level statement is not picked up by ENC

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#44196]
![image](https://user-images.githubusercontent.com/10522107/81750452-5e6b0800-9462-11ea-8706-016d4135496e.png)

Debugger simply ignores new code afterwards. It is quite possible that "rude" edits detection isn't quite working.

Relates to https://github.com/dotnet/roslyn/issues/43563 (test plan for "top-level statements")

**Comment by tmat** [comment]
Nothing about top-level code will work without explicit support for the feature in Edit and Continue. It is also possible that EnC is broken when editing non-top-level code (e.g. a top-level class) when top-level code is present.

**Comment by AlekseyTs** [comment]
Indeed, editing regular code in the file with top-level statements has the same issue.

**Comment by AlekseyTs** [comment]
@tmat It is quite possible that EnC is broken for RoslynDeployment. I ran into similar issues with a program that doesn't have top-level statements.

**Comment by tmat** [comment]
Just occurred to me... are you using master-vs-deps? EnC does not work in master since it requires the latest VS bits.

**Comment by AlekseyTs** [comment]
No, I am trying bits from features/SimplePrograms branch, which is almost up to date with master.

**Comment by tmat** [comment]
That explains. For end-to-end testing you'd need 1) branch off of master-vs-deps 2) make changes in the IDE to support statement diffing in top-level code. 
I can do the IDE work once your branch merges to master. The compiler side work only needs unit testing similar to what's done for other features in `Compilers\CSharp\Test\Emit\Emit\EditAndContinue\EditAndContinueTests.cs`.

**Comment by AlekseyTs** [comment]
@tmat Just an FYI, the feature has been merged into master

**Comment by tmat** [comment]
Thanks!

**Comment by pranavkm** [comment]
We are looking at expanding hot reload to support changes to an [ASP.NET Core's bootstrapping code](https://github.com/dotnet/aspnetcore/issues/32648). Starting in 6.0, we're going to push a minimal hosting solution that relies on top level statements - https://github.com/dotnet/aspnetcore/issues/30354. Fixing this would unblock this scenario.

FYI @davidfowl
